tests/pipelines/test_pipelines_common.py

```diff
@@ -138,7 +138,8 @@ def get_tiny_tokenizer_from_checkpoint(checkpoint):
         return tokenizer
     logger.info("Training new from iterator ...")
     vocabulary = string.ascii_letters + string.digits + " "
-    tokenizer = tokenizer.train_new_from_iterator(vocabulary, vocab_size=len(vocabulary), show_progress=False)
+    if "Whisper" not in tokenizer.__class__.__name__:
+        tokenizer = tokenizer.train_new_from_iterator(vocabulary, vocab_size=len(vocabulary), show_progress=False)
```

Comment:
This is a bit hacky, but retraining the whisper tokenizer leads to an infinite recursino later on. Small reproduction:

```python
import string
from transformers import WhisperTokenizerFast

tokenizer = WhisperTokenizerFast.from_pretrained("openai/whisper-small")
vocabulary = string.ascii_letters + string.digits + " "
tokenizer = tokenizer.train_new_from_iterator(vocabulary, vocab_size=len(vocabulary), show_progress=False)
tokenizer.all_special_ids
```

The recursion is because we try to compute unknown token id, so we convert unknown token to id, but this lookup fails (because of the retraining I assume), so we try to return unknown token id, ...

Reply:
This should be fixed with #21250, we are probably going to set `unk_token = <|endoftext|>`. WDYT?

Reply:
This issue is more about converting the unknown token `''` to id, and not the other way round. But perhaps we could do some sort of override in the fast tokenizer as well. Another thing is that `unk_token` is `"<|endoftext|>"` by default, it's just the checkpoints configuration that uses `''`, so probably not something we want to assume in all cases?